$MirOS: ports/security/ekeyd/patches/patch-daemon_ekey-rekey_in,v 1.2 2009/11/22 19:32:32 tg Exp $
--- host/ekey-rekey.in.orig	Fri Jul 22 11:42:55 2011
+++ host/ekey-rekey.in	Sun Jan 15 13:35:42 2012
@@ -8,7 +8,7 @@ usage() {
     cat <<EOF >&2
 ekey-rekey: Utility to re-key an Entropy Key's Long-term-key
 Usage:
-    ekey-rekey [-d|--device DEVICENODE] [SERIAL [MASTERKEY]]
+    ekey-rekey [{-d|--device} DEVICENODE [SERIAL [MASTERKEY]]]
 EOF
 }
 
@@ -44,10 +44,8 @@ sysfs_find_ekey_dev () {
     fi
 }
 
-if test $# -ge 1; then
-
 optloop=1
-while test ${optloop} = 1; do
+while test ${optloop} = 1 && test $# -ge 1; do
     case x$1 in
     x-d|x--device)
 	DEVICE=$2
@@ -82,8 +80,6 @@ while test ${optloop} = 1; do
     esac
 done
 
-fi
-
 # first open parameter must be the serial number
 SERIAL=$1
 
@@ -92,7 +88,12 @@ if [ "x${SERIAL}" = "x" ]; then
     FOUND_EKEY=$(sysfs_find_ekey)
 
     if [ "x${FOUND_EKEY}" = "x" ];then
-	echo >&2 "No Entropy Key could be found. Is it connected?"
+	if test -d "$DEVDIR/."; then
+		echo >&2 "No Entropy Key could be found. Is it connected?"
+	else
+		echo >&2 "No Linux sysfs, you must specify the serial number."
+		usage
+	fi
 	exit 2
     fi
     FOUND_EKEY_DEV=$(sysfs_find_ekey_dev ${FOUND_EKEY})
